lib/bap_disasm/bap_disasm_rec.ml

```diff
@@ -223,12 +142,9 @@ let is_barrier s mem insn =
   has_jump (ok_nil (s.lift mem insn))
 
 let update s mem insn dests : stage1 =
-  let s = {s with valid = Set.add s.valid (Memory.min_addr mem)} in
+  let s = { s with visited = Visited.add s.visited mem } in
   if is_barrier s mem insn then
-    let dests = List.map dests ~f:(fun d -> match d with
-        | Some addr,kind when Memory.contains s.base addr -> d
-        | _,kind -> None, kind) in
-    update_dests s dests mem;
+    let () = update_dests s mem dests in
```

Comment:
It also looks like that the implementation of the `is_barrier` function diverged from its name. Originally, we were classifying functions by using the is_barrier predicate provided by the disassembler, so that we were either pushing the fallthrough address or the address of an instruction. Many eons later, we basically started to classify them by whether they employ normal control flow with the fallthrough edge only, or it is a jump instruction, which may or may not have the fallthrough. 

Long story short, let's rename the `is_barrier` function to `is_jump`